inc/features/blocks/personalization/components/variant-title.js

```diff
@@ -0,0 +1,47 @@
+const { useSelect } = wp.data;
+const { __ } = wp.i18n;
+
+// Component for fetching and displaying the variant title string.
+const VariantTitle = ( { variant } ) => {
+	const audience = useSelect( select => {
+		return select( 'audience' ).getPost( variant.attributes.audience );
+	}, [ variant.attributes.audience ] );
+
+	const isLoading = useSelect( select => {
+		return select( 'audience' ).getIsLoading();
+	}, [] );
+
+	if ( variant.attributes.fallback ) {
+		return __( 'Fallback', 'altis-experiments' );
+	}
+
+	if ( ! variant.attributes.audience ) {
+		return __( 'Select audience', 'altis-experiments' );
+	}
+
+	const status = ( audience && audience.status ) || 'draft';
+	const title = audience && audience.title && audience.title.rendered;
+
+	// Audience is valid and has a title.
+	if ( status !== 'trash' && title ) {
+		return audience.title.rendered;
+	}
+
+	// Audience has been deleted.
+	if ( status === 'trash' ) {
+		return __( '(deleted)', 'altis-experiments' );
```

Comment:
Should this be title + deleted? 🤔

Reply:
Mm, lets leave it for now, that could be quite lengthy